map/src/map/layers/FavoriteLayer.js

```diff
@@ -29,6 +29,20 @@ const FavoriteLayer = () => {
         }
     }, [ctx.zoomToFavGroup]);
 
+    useEffect(() => {
+        if (!ctx.loginUser) {
+            // If there is no logged-in user, remove all layers except tiles.
+            map.eachLayer((layer) => {
+                if (!layer.options.attribution) {
+                    map.removeLayer(layer);
+                }
+            });
+
+            // Clear the cache of favorites.
+            localStorage.removeItem(FAVORITE_STORAGE);
+        }
+    }, [ctx.loginUser]);
```

Comment:
Please do cleanup for Favorites layer only. Do not remove tracks/navigation/weather/etc